### DISCUSSION TITLE
Panel faintly showing colors when it should show full black

### DISCUSSION THREAD

**Discussion by samvoigt** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#280] [Q&A]
Hello,
When displaying a "black" screen, I'm noticing that some pixels are still very dimly showing color (this was taken in low light so it's a bit exaggerated, but it's noticeable):
![IMG_3277](https://user-images.githubusercontent.com/6341318/166499744-2168bfde-76ae-42e5-97fa-fb1f1a5232ba.jpeg)

Some of the "off" pixels are red, some are green, some are red and green. I don't see any blue showing up. This happens after the call to `dma_display->begin();`, and remains regardless of what I do (e.g. filling the screen with black). The panels are fully black when powered on before this call. Anything else I draw seems to display fine, but I can see the red/green poking through. 

When I have these same panels hooked up to a Raspberry Pi/Adafruit Hat (same setup as below), I don't have this problem.

Setup:
- 4 of these panels: https://www.amazon.com/gp/product/B079JSKF21
- ESP32 Trinity board
- 150W power supply: https://www.amazon.com/gp/product/B01HJA3OUG

What I've tried so far:
- Changing `latch_blanking`. This doesn't work, and the problem gets MUCH worse (e.g. the off pixels get brighter) if it's anything but 1.
- Changing the brightness. This doesn't work, same as above as brightness approaches 0.
- Changing `min_refresh_rate`: no effect
- Changing `clk_speed`: no effect
- No difference between double buffering and not
- No difference changing `clkphase`

Any help would be much appreciated, thanks!

**Comment by mrcodetastic** [comment]
Chuck the supply voltage. Might be too high.

Grounding on ESP and panels may not be right.

**Comment by ergindemir** [comment]
I have the same issue with a 64x64 p3 panel with fm6126 chip.

**Comment by board707** [comment]
Try to use the level shifter (3.3 -> 5v) on OE line , and better on each signal lines, there are some panels that needs it

**Reply by ergindemir** [reply]
I checked my working and malfunctioning panels. Both use the same logic buffer, 74hc245. It's highly unlikely same buffer malfunctions on other device.

**Comment by ergindemir** [comment]
I have tested the 64x64 p3 panel with 4-4.2V power supply, the issue still persists. I think upon this observation we can conclude that it's not a logic buffer issue.

**Comment by mrcodetastic** [comment]
Can you try the old legacy branch of this library and see if you still have the same issues?

https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-I2S-DMA/tree/legacy-1.2.4

**Comment by zachtos** [comment]
I have the same issue. It's a show stopper for production for us.  The pixels just idle with some minor low power color never being fully black after the library starts.  The FM6126A will not turn black, but the exact same panel with nothing different other than uses ICN2038S (ICN2073) works as expected with same code.  This is using the matrix S3 from adafruit, so level shifting is already 5V.  The 5V PSU direct to display does not affect the issue, and every configuration does not matter.  I can quickly swap back in the same panel w/ different driver and it's resolved (after swap the driver in code).  Legacy library would not compile with ESP32-S3.  An odd thing is that it seems inversely proportional to the brightness setting.  Lower brightness and brighter 'off LED', but high brightness and lower brightness of LED in off position, something is really messed up.  It uses way too much power when all the LED never shut off.  I am willing to pay for help.

https://www.alibaba.com/product-detail/P2-5-Indoor-RGB-Full-Color_1600802741272.html?spm=a2756.trade-list-buyer.0.0.13c776e9oegvLY

**Reply by board707** [reply]
> (after swap the driver in code).

What do you means as swapping driver in code? Does the panels that show the dim pixels and the panles that works fine run with different library settings?

**Reply by w531t4** [reply]
> I have the same issue. It's a show stopper for production for us. The pixels just idle with some minor low power color never being fully black after the library starts. The FM6126A will not turn black, but the exact same panel with nothing different other than uses ICN2038S (ICN2073) works as expected with same code. This is using the matrix S3 from adafruit, so level shifting is already 5V. The 5V PSU direct to display does not affect the issue, and every configuration does not matter. I can quickly swap back in the same panel w/ different driver and it's resolved (after swap the driver in code). Legacy library would not compile with ESP32-S3. An odd thing is that it seems inversely proportional to the brightness setting. Lower brightness and brighter 'off LED', but high brightness and lower brightness of LED in off position, something is really messed up. It uses way too much power when all the LED never shut off. I am willing to pay for help.
> 
> https://www.alibaba.com/product-detail/P2-5-Indoor-RGB-Full-Color_1600802741272.html?spm=a2756.trade-list-buyer.0.0.13c776e9oegvLY

i see the same behavior on my panel, with adafruit matrix s3. did you ever find a fix? If i switch back to the stock adafruit test firmware, the display works just fine.
